### PR TITLE
Initial attempt at adding Maven support.

### PR DESCRIPTION
**Pull request by claymccoy**
I'd love to get some feedback on this, but don't merge it yet. It needs a fix to sdm before it will work correctly (atomist/sdm#490), and it needs more testing after that.

Moved stuff out of node support that wasn't directly tied to node.
Use goal implementation push tests to differentiate node and maven goals.
Made executeReleaseVersion generic rather than node specific by allowing an incrementPatchCmd to be specified.

### CODE REVIEW

File: src/machine/dockerSupport.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    sdm.addGoalImplementation(
+            "dockerRelease",
+            ReleaseDockerGoal,
+            executeReleaseDocker(sdm.configuration.sdm.projectLoader,
+                DockerReleasePreparations,
+                {
+                    ...sdm.configuration.sdm.docker.hub as DockerOptions,
+                }),
+            {
+                pushTest: allSatisfied(HasDockerfile),
```
Comment (ddgenome) [inline src/machine/dockerSupport.ts]: No need for `allSatisfied` if there is only one predicate.

File: src/machine/k8Support.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    const ns = namespaceFromGoal(goal);
+    const ingress = ingressFromGoal(goal.repo.name, ns);
+    return createKubernetesData(
+        goal,
+        {
+            name: goal.repo.name,
+            environment: configuration.environment.split("_")[0],
+            port: 2866,
+            ns,
+            imagePullSecret: "atomistjfrog",
```
Comment (ddgenome) [inline src/machine/k8Support.ts]: Seems we would not need an imagePullSecret.

File: test/machine/k8SupportTest.ts [hunk 3]
```diff
@@ -15,12 +15,9 @@
  */
 
 import * as assert from "power-assert";
+import {ingressFromGoal} from "../../src/machine/k8Support";
```
Comment (ddgenome) [inline test/machine/k8SupportTest.ts]: Spacing

Comment (claymccoy) [reply test/machine/k8SupportTest.ts]: Yep, IntelliJ seems to add dependencies like this, and lint:fix doesn't do anything about it.

File: src/machine/k8Support.ts (hunk 2 above)
Comment (claymccoy) [reply src/machine/k8Support.ts]: Why is that needed for the node projects? I don't remember what that was for.

I need to conditionally change the port to 8080, maybe use the IsNode/IsMaven push tests,  or maybe I could do something more relevant like parse the DockerFile for exposed ports.

Comment (claymccoy) [reply src/machine/k8Support.ts]: I've left imagePullSecret in. It was already there.

**Pull request closed without merging**